rules/rules.js

```diff
@@ -403,19 +403,24 @@ const getTriggeredData = function (input) {
 
   // Only with event data (for Item, Thing & Channel triggers)
   if (event) {
+    try {
+      data.itemName = event.getItemName();
+    } catch (e) {
+      // do nothing
+    }
     switch (Java.typeName(event.class)) {
+      case 'org.openhab.core.items.events.GroupItemCommandEvent':
```

Comment:
Instead of the try/catch block, what about setting it in the case statement first without a break, so the statement falls through to the group/item logic ? 
```
      case 'org.openhab.core.items.events.ItemCommandEvent':
        data.itemName = event.getItemName();
        // no break here
      case 'org.openhab.core.items.events.GroupItemCommandEvent':
        data.itemName = event.getItemName();
        data.eventType = 'command';
        data.triggerType = 'ItemCommandTrigger';
        break;
```

Reply:
I created the try catch block to add `itemName` every time it is available, but I am also comfortable with setting `itemName` in the switch statement.

Reply:
I'm ok either way, if i actually write the whole case statement out, it does get a little verbose, so maybe your way is better :-)  


```javascript
      case 'org.openhab.core.items.events.ItemCommandEvent':
         data.itemName = event.getItemName();
      case 'org.openhab.core.items.events.GroupItemCommandEvent':
        data.eventType = 'command';
        data.triggerType = 'ItemCommandTrigger';
        break;
      case 'org.openhab.core.items.events.ItemStateChangedEvent':
        data.itemName = event.getItemName();
      case 'org.openhab.core.items.events.GroupItemStateChangedEvent':
        data.eventType = 'change';
        data.triggerType = 'ItemStateChangeTrigger';
        break;
      case 'org.openhab.core.items.events.ItemStateEvent':
        data.itemName = event.getItemName();
      case 'org.openhab.core.items.events.GroupItemStateEvent':
        data.eventType = 'update';
        data.triggerType = 'ItemStateUpdateTrigger';
```